extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad-network-doubleclick-impl.js

```diff
@@ -1552,4 +1556,122 @@ describes.sandboxed('amp-ad-network-doubleclick-impl', {}, () => {
       });
     });
   });
+
+  describe('#correlator clear experiment', () => {
+    let onVisibilityChangedHandler;
+    let visabilityState;
+
+    beforeEach(() => {
+      onVisibilityChangedHandler = null;
+      visabilityState = VisibilityState.PAUSED;
+      return createIframePromise().then(fixture => {
+        const viewer = {
+          isVisible: () => true,
+          onVisibilityChanged: handler => {
+            onVisibilityChangedHandler = handler;
+          },
+          getVisibilityState: () => visabilityState,
+          whenFirstVisible: () => Promise.resolve(),
+        };
+        sandbox.stub(Services, 'viewerForDoc').returns(viewer);
+        element = createElementWithAttributes(fixture.doc, 'amp-ad', {
+          type: 'doubleclick',
+          height: '250',
+          width: '320',
+        });
+        document.body.appendChild(element);
+        impl = new AmpAdNetworkDoubleclickImpl(element);
+        impl.win.ampAdPageCorrelator = 12345;
+        delete impl.win['dbclk_a4a_viz_change'];
+      });
+    });
+
+    it('clears if in experiment', () => {
+      impl.win.ampAdPageCorrelator = 12345;
+      forceExperimentBranch(impl.win, CORRELATOR_CLEAR_EXP_NAME,
+          CORRELATOR_CLEAR_EXP_BRANCHES.EXPERIMENT);
+      impl.buildCallback();
+      expect(onVisibilityChangedHandler).to.be.ok;
+      onVisibilityChangedHandler();
+      expect(impl.win.ampAdPageCorrelator).to.not.be.ok;
+      expect(isInExperiment(element, CORRELATOR_CLEAR_EXP_BRANCHES.CONTROL))
+          .to.be.false;
+      expect(isInExperiment(element, CORRELATOR_CLEAR_EXP_BRANCHES.EXPERIMENT))
+          .to.be.true;
+    });
+
+    it('does not clear if in control', () => {
+      forceExperimentBranch(impl.win, CORRELATOR_CLEAR_EXP_NAME,
+          CORRELATOR_CLEAR_EXP_BRANCHES.CONTROL);
+      impl.buildCallback();
+      expect(onVisibilityChangedHandler).to.be.ok;
+      onVisibilityChangedHandler();
+      expect(isInExperiment(element, CORRELATOR_CLEAR_EXP_BRANCHES.CONTROL))
```

Comment:
You don't actually check the value of impl.win.ampAdPageCorrelator in this test.